server/src/main/java/org/opensearch/common/compress/ZstdCompressor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.compress;
+
+import com.github.luben.zstd.RecyclingBufferPool;
+import com.github.luben.zstd.ZstdInputStreamNoFinalizer;
+import com.github.luben.zstd.ZstdOutputStreamNoFinalizer;
+import org.opensearch.common.bytes.BytesReference;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+
+/**
+ * {@link Compressor} implementation based on the ZSTD compression algorithm.
+ *
+ * @opensearch.internal
+ */
+public class ZstdCompressor implements Compressor {
+    // An arbitrary header that we use to identify compressed streams
+    // It needs to be different from other compressors and to not be specific
+    // enough so that no stream starting with these bytes could be detected as
+    // a XContent
+    private static final byte[] HEADER = new byte[] { 'Z', 'S', 'T', 'D', '\0' };
```

Comment:
readability nit: 
```suggestion
    public static final String NAME = "ZSTD";
    private static final byte[] HEADER =NAME.getBytes();
```

Reply:
Well I just wanted to do it as it done for `DEFLATE { 'D', 'F', 'L', '\0' } ` not sure about `\0`